### DISCUSSION TITLE
Is there any example for distributed training a customized NER model in Spacy 3.0?

### DISCUSSION THREAD

**Discussion by gaohuan2015** [explosion/spaCy#8198] [Help: Coding & Implementations] [answered]
Hi Community

I use Spacy 3.0 to train a customized NER model. The data of my work is very large and I want to use a distributed way to train the customized NER model in Spacy framework. I know spacy-ray can do a similar thing, but the doc is not very clear. My question is how to config the cluster or is there any example for training spacy in a distributed way?

**Comment by adrianeboyd** [comment, marked as answer]
Hi, sorry for the late reply! If you haven't seen it, there's an example `spacy-ray` project here:

https://github.com/explosion/projects/tree/v3/integrations/ray

Get started with `spacy project clone integrations/ray`.

The example config is for training a tagger/parser on a UD corpus instead of for NER, but the basic project setup and training steps should be very similar.

The example project does not include many details/customizations related to the ray cluster configuration, you'll need to refer to the ray docs for more information about that. Be aware that `spacy-ray` currently requires a slightly older version, ray v0.8.x.
